x-pack/plugins/security_solution/public/app/home/global_header/index.tsx

```diff
@@ -53,22 +53,24 @@ export const GlobalHeader = React.memo(() => {
   const { href, onClick } = useAddIntegrationsUrl();
 
   useEffect(() => {
-    setHeaderActionMenu((element) => {
-      const mount = toMountPoint(<OutPortal node={portalNode} />, {
-        theme,
-        i18n: kibanaServiceI18n,
+    if (setHeaderActionMenu) {
```

Comment:
If we make `params` optional like mentioned in [this comment](https://github.com/elastic/kibana/pull/176656/files#r1563249378) maybe we can set `setHeaderActionMenu` as a `noop` and avoid this check and maybe even the removal of `onAppLeave` in the below files?

I'm not super familiar with the plugin registration logic and how that works the management UI's, so if this might not be a good suggestion please ignore!